source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -594,9 +594,37 @@ namespace
         return destNode;
     }
 
-    void writeMaterialElements(const PvtNode* node, NodePtr mxNode, const string& nodeName, DocumentPtr doc, const RtWriteOptions* writeOptions)
+    void createMaterialNode(PvtNode* node, NodePtr mxNode, DocumentPtr doc)
```

Comment:
Creates the material node for a provided surface shader node if one does not already exist.